cci_nepal/pipeline/dummy_data/model_complete.py

```diff
@@ -0,0 +1,367 @@
+    "Cotton Towel",
+    "Laundry Soap",
+    "Tooth Brush and Paste",
+    "Sanitary Pad",
+    "Ladies Underwear",
+    "Torch Light",
+    "Whistle Blow",
+    "Nail Cutter",
+    "Hand Sanitizer",
+    "Liquid Chlorine",
+]
+
+y_basic = df_clean.loc[:, nfri_basic]
+
+
+y_non_basic = df_clean.loc[:, nfri_non_basic]
+# -
+
+# #### Fitting the Linear Regression Model for both NFRI below ####
+
+# First we will create the train and test dataset. And after that, we will fit on the train set and predict on test.
+
+X_train_basic, X_test_basic, y_train_basic, y_test_basic = train_test_split(X, y_basic)
+(
+    X_train_non_basic,
+    X_test_non_basic,
+    y_train_non_basic,
+    y_test_non_basic,
+) = train_test_split(X, y_non_basic)
+
+
+model_basic.fit(X_train_basic, y_train_basic)
+model_non_basic.fit(X_train_non_basic, y_train_non_basic)
+
+# #### Predicting the Model for Test Set so as to calculate accuracy below ####
+
+test_predictions_basic = model_basic.predict(X_test_basic)
+test_predictions_non_basic = model_non_basic.predict(X_test_basic)
+
+
+# +
+test_predictions_labels_basic = [
+    [ddu.numeric_score_transformer(i) for i in nested]
+    for nested in test_predictions_basic
+]
+test_predictions_items_wise_basic = list(
+    map(list, zip(*test_predictions_labels_basic))
+)  # Transforming the above list
+
+test_predictions_labels_non_basic = [
+    [ddu.numeric_score_transformer(i) for i in nested]
+    for nested in test_predictions_non_basic
+]
+test_predictions_items_wise_non_basic = list(
+    map(list, zip(*test_predictions_labels_non_basic))
+)  # Transforming the above list
+# -
+
+
+# #### Accuracy of the Model for each NFRI item ####
+
+# The below section will display the accuracy of the model (using the Categorical labels mapped from numerical score) for each NFRI item.
+
+# +
+accuracy_list_basic = []
+for i in range(0, len(nfri_basic)):
+    accuracy_list_basic.append(
+        accuracy_score(test_predictions_items_wise_basic[i], y_test_basic.iloc[:, i])
+    )
+
+logging.info("Accuracy Per NFRI Basic Item")
+logging.info(
+    pd.DataFrame(accuracy_list_basic, y_test_basic.columns, columns=["Accuracy"])
+)
+
+
+# +
+accuracy_list_non_basic = []
+for i in range(0, len(nfri_non_basic)):
+    accuracy_list_non_basic.append(
+        accuracy_score(
+            test_predictions_items_wise_non_basic[i], y_test_non_basic.iloc[:, i]
+        )
+    )
+
+logging.info("Accuracy Per NFRI Non Basic Item")
+logging.info(
+    pd.DataFrame(
+        accuracy_list_non_basic, y_test_non_basic.columns, columns=["Accuracy"]
+    )
+)
+
+# -
+
+# #### Analysing the Model Coefficients ####
+
+# In this section, we will analyse the Model Coefficients of the fitted Linear Regression Model. For now, the Model here is fitted using the whole dataset.
+
+model_basic.fit(X, y_basic)
+model_non_basic.fit(X, y_non_basic)
+
+
+coefficient_scores = model_basic.coef_[0]
+plt.bar([x for x in range(len(coefficient_scores))], coefficient_scores)
```

Comment:
Although you could have a function in pipeline* that builds a plot depending on certain inputs, I think actually plotting of data should be in analysis.

*Maybe it would be utils rather than pipeline though?